### PR TITLE
Refactoring plone client to use axios replacing superagent

### PR DESCRIPTION
**Pull request by hemant-hc**
This PR introduces: 

1. Refactoring API ts file to use axios replacing superagent.
2. Fixes for actions and their tests accordingly.

### CODE REVIEW

**Comment by hemant-hc** [comment]
@sneridagh this PR is good to go now; NextJS demo should be unblocked.

As for adding new tests related to `API.ts`, ~I am working on a new PR to add those (keeping it separate to unblock this PR first).~ Added the changes in this PR itself.

My plan is to:
1. mock and spy on (`vi.spyOn`) axios.create
2. test that request and response interceptors are set
3. test that `AxiosInstance` is being called with the correct parameters (method, path, headers, params etc) when `apiRequest` is called

Actual network related tests for get/post/patch/delete are already covered under the tests for the Content action anyway.

File: src/API.ts [hunk 1]
```diff
@@ -24,103 +21,73 @@ function getBackendURL(apiPath: string, path: string) {
...
+): Promise<any> {
+  const instance = axios.create();
+
+  if (options.raw) {
+    instance.interceptors.response.use(undefined, _handleError);
+  } else {
+    instance.interceptors.response.use(_handleResponse, _handleError);
   }
+
+  return instance.request(axiosConfigAdapter(method, path, options));
```
Comment (hemant-hc) [inline src/API.ts:92]: Changed `instance()` to `instance.request()` to be able to mock easily in tests.

File: vitest.config.ts [hunk 2]
```diff
@@ -8,5 +8,21 @@ export default defineConfig({
...
+          to avoid CORS issue.
+          
+          This is because JSDom environment sets the 
+          origin header (axios cannot control it on its own, just like in a 
+          browser) automatically to the url specified here. 
+          
+          If this url is not provided then vitest sets the default value as 
+          localhost:3000 which causes CORS error with the test server.
+        */
+        url: 'http://localhost:55001',
```
Comment (hemant-hc) [inline vitest.config.ts:24]: CORS error appears in tests when superagent is replaced with axios in `resetFixtures.ts`. Not sure why that would be, but I inspcted the API requests being made with fiddler and requests made with superagent did not contain the `Origin` header (used to enforce CORS) while requests made with axios did contain this header.

Setting the window URL to `localhost:55001` for JSDom environment fixes the issue.

Comment (sneridagh) [reply vitest.config.ts:24]: Ok, I understand now the CORS error happened in `resetFixtures.ts`.

**Review by sneridagh — APPROVED** [review]

**Pull request merged**